sql/src/main/java/io/crate/executor/transport/ShardUpdateRequest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.executor.transport;
+
+import io.crate.Constants;
+import io.crate.planner.symbol.Reference;
+import io.crate.planner.symbol.Symbol;
+import org.elasticsearch.action.DocumentRequest;
+import org.elasticsearch.action.support.single.instance.InstanceShardOperationRequest;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.uid.Versions;
+import org.elasticsearch.index.mapper.Uid;
+import org.elasticsearch.index.shard.ShardId;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ShardUpdateRequest extends InstanceShardOperationRequest<ShardUpdateRequest> implements DocumentRequest<ShardUpdateRequest> {
+
+    private String type;
+    private String id;
+    @Nullable
+    private String routing;
+    private long version = Versions.MATCH_ANY;
+    private int retryOnConflict = 0;
+    private Map<Reference, Symbol> assignments;
```

Comment:
maybe you could just use `Map<String, Symbol`> or `Map<ColumnIdent, Symbol>`  I think the tableIdent and typeInfo inside the Reference isn't needed/redundant